aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -408,13 +424,72 @@ impl AptosVM {
         self.execute_module_initialization(
             &mut session,
             gas_status,
-            modules,
+            &self.deserialize_module_bundle(modules)?,
             &[txn_data.sender()],
         )?;
 
         self.success_transaction_cleanup(session, gas_status, txn_data, log_context)
     }
 
+    /// Resolve a pending code publish request registered via the NativeCodeContext.
+    fn resolve_pending_code_publish<S: MoveResolverExt>(
+        &self,
+        session: &mut SessionExt<S>,
+        gas_status: &mut GasStatus,
+    ) -> VMResult<()> {
+        if let Some(PublishRequest {
+            destination,
+            bundle,
+            expected_modules,
+            check_compat,
+        }) = NativeCodeContext::extract_publish_request(session)
+        {
+            // TODO: unfortunately we need to deserialize the entire bundle here to handle
+            // `init_module` and verify some deployment conditions, while the VM need to do
+            // the deserialization again. Consider adding an API to MoveVM which allows to
+            // directly pass CompiledModule.
+            let modules = self.deserialize_module_bundle(&bundle)?;
+
+            // Validate the module bundle
+            self.validate_publish_request(&modules, expected_modules)?;
+
+            // Publish the bundle
+            if check_compat {
+                session.publish_module_bundle(bundle.into_inner(), destination, gas_status)?
+            } else {
+                session.publish_module_bundle_relax_compatibility(
+                    bundle.into_inner(),
+                    destination,
+                    gas_status,
+                )?
+            }
+
+            // Execute initializers
```

Comment:
nit: Comment is not needed/redundant with the code.

Reply:
Sorry I do not buy in this code style.

Reply:
My main point is that the comments are not super helpful as they don't explain why the code does something. It barely explains what it does. In general, comments should explain if the code does something expected or is too complex.

Reply:
I totally understand your point and thanks for the feedback. But I do not believe in this rule of code style, and have actively fought against this particular detail e.g. at Google which has very stringent style guidelines. So I respectfully reject your change request.

TL;DR I found code more readable if there are english comments explaining the flow. What appears redundant today is often not any longer if the code evolves; in fact it might be necessary in the future. For example, `Validate the module bundle` might be one simple step today but multiple in the future.